pkg/logs/processor/processor.go

```diff
@@ -34,30 +52,88 @@ type Processor struct {
 	diagnosticMessageReceiver diagnostic.MessageReceiver
 	mu                        sync.Mutex
 	hostname                  hostnameinterface.Component
+	config                    pkgconfigmodel.Reader
+	configChan                chan failoverConfig
+	failoverConfig            failoverConfig
 
 	// Telemetry
 	pipelineMonitor metrics.PipelineMonitor
 	utilization     metrics.UtilizationMonitor
 	instanceID      string
 }
 
-// New returns an initialized Processor.
-func New(inputChan, outputChan chan *message.Message, processingRules []*config.ProcessingRule,
+// New returns an initialized Processor with config support for failover notifications.
+func New(config pkgconfigmodel.Reader, inputChan, outputChan chan *message.Message, processingRules []*config.ProcessingRule,
 	encoder Encoder, diagnosticMessageReceiver diagnostic.MessageReceiver, hostname hostnameinterface.Component,
 	pipelineMonitor metrics.PipelineMonitor, instanceID string) *Processor {
 
-	return &Processor{
+	p := &Processor{
+		config:                    config,
 		inputChan:                 inputChan,
 		outputChan:                outputChan, // strategy input
 		processingRules:           processingRules,
 		encoder:                   encoder,
+		configChan:                make(chan failoverConfig, 1),
 		done:                      make(chan struct{}),
 		diagnosticMessageReceiver: diagnosticMessageReceiver,
 		hostname:                  hostname,
 		pipelineMonitor:           pipelineMonitor,
 		utilization:               pipelineMonitor.MakeUtilizationMonitor(metrics.ProcessorTlmName, instanceID),
 		instanceID:                instanceID,
 	}
+
+	// Initialize cached failover config
+	p.updateFailoverConfig()
+
+	// Register for config change notifications
+	if config != nil {
+		config.OnUpdate(p.onLogsFailoverSettingChanged)
+	}
+
+	return p
+}
+
+// onLogsFailoverSettingChanged is called when any config value changes
+func (p *Processor) onLogsFailoverSettingChanged(setting string, _ pkgconfigmodel.Source, _, _ any, _ uint64) {
+	// Only update if the changed setting affects failover configuration
+	failoverSettingChanged := setting == "multi_region_failover.failover_logs" || setting == "multi_region_failover.logs_service_allowlist" || setting == "multi_region_failover.logs_min_level"
+	if failoverSettingChanged {
+		p.updateFailoverConfig()
+	}
+}
+
+// updateFailoverConfig sends the updated config to the processor to update
+func (p *Processor) updateFailoverConfig() {
+	if p.config == nil {
+		return
+	}
+
+	var conf failoverConfig
```

Comment:
I know you did it some lines below, but since the default value for `conf.failoverMinLevel` is `-1` and not `0`, consider setting it at the creation time.

```
conf := failoverConfig{
  failoverMinLevel: -1,
  isFailoverActive: p.config.GetBool("multi_region_failover.failover_logs"),
}
```
It will avoid a potential mistake in a future refactoring.

Another way would have to have a constructor attached to `failoverConfig` but I don't think it's necessary here.

Reply:
Makes sense, intention seems clearer this way.